File: ps/SW/runway.py
import sys
sys.stdin = open("runway_input.txt")

T = int(input())

def recurse(array, X, before=0):
    count = 0
    if len(array) == count:
        return True
    if not before:
        before = array[0]
    for idx, height in enumerate(array):
        if before == height:
            count += 1
            continue
        elif height == before + 1 and count >= X:
            return recurse(array[idx:], X)
        elif height == before - 1 and array[idx:idx+X] == [array[idx]]*(X):
            return recurse(array[idx+X:], X, height)
        elif height == before -1 and array[idx:] == [array[idx]] *len(array[idx:]) and len(array[idx:]) >= X:
            return True
        else:
            return False
    return True

for tc in range(1, T+1):
    N, X = map(int, input().split())
    grid = [list(map(int, input().split())) for _ in range(N)]
    count = 0
    for row in grid:
        count += recurse(row, X)
    turn = [[grid[r][c] for r in range(N)] for c in range(N)]
    for row in turn:
        count += recurse(row, X)
    print("#{} {}".format(tc,count))

File: ps/SW/runway.cpp
#include <iostream>
#include <cstdio>
using namespace std;
int grid[20][20] = {};
int N, X;

bool check(int arr[]) {
    int before = arr[0];
    int count = 0;
    for (int i = 0; i < N; i++) {
        if (before == arr[i]) {
            count ++;
        } else if (before+1 == arr[i]) {
            if (count >= X) {
                count = 1;
                before = arr[i];
            } else {
                return false;
            }
        } else {
            if (i+X > N) return false;  //gives me pass ;;
            int checker = before - 1;
            for (int j = 0; j < X; j++) {
                if (arr[i+j] != checker) {
                    return false;
                }
            }
            count = -X+1;  // -x+1 ->30 48
            before = arr[i];
        } 
    } 
    return true;
}


int main() {
    freopen("runway_input.txt", "r", stdin);
    int T;
    scanf("%d", &T);
    for (int tc = 0; tc < T; tc++) {
        scanf("%d %d", &N, &X);
        for (int y = 0; y < N; y++)
            for (int x = 0; x <N; x++)
                scanf("%d", &grid[y][x]);
        
        int result = 0;
        for (int y = 0; y < N; y++) {
            result += check(grid[y]);
        }
        for (int y = 0; y < N; y++) {
            int arr[20] = {};
            for (int x = 0; x < N; x++) {
                arr[x] = grid[x][y];
            }
            result += check(arr);
        }
        printf("#%d %d\n", tc+1, result);

    }
    return 0;
}
